### ISSUE TITLE
[Bug]: `alter table` cannot add column of type `enum`

### ISSUE THREAD

**Issue by aronchanisme** [matrixorigin/matrixone#15176]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

main

### Commit ID

6e2ff3c8c

### Other Environment Information

```Markdown
- Hardware parameters: 8c 16g
- OS type: Debian 11
- Others:
uname -a
Linux test0 5.10.0-28-amd64 #1 SMP Debian 5.10.209-2 (2024-01-31) x86_64 GNU/Linux
```

### Actual Behavior

In MO, `alter table` cannot add column of type `enum` as MO throws an error indicating syntax error.

```bash
mysql> create table t3 (id int);
Query OK, 0 rows affected (0.02 sec)

mysql> insert into t3 values (1);
Query OK, 1 row affected (0.01 sec)
mysql> use test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> alter table t3 add column name enum ('A','B','C');
ERROR 1105 (HY000): You have an error in your SQL syntax; check the manual that corresponds to your MatrixOne server version for the right syntax to use. syntax error at line 4 column 2 near " 
)";
```

### Expected Behavior

While in MySQL 8.0, this works as expected.

```bash
mysql> create table t3 (id int);
Query OK, 0 rows affected (0.03 sec)

mysql> insert into t3 values 1;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '1' at line 1
mysql> insert into t3 values (1);
Query OK, 1 row affected (0.01 sec)

mysql> alter table t3 add column name enum ('A','B','C');
Query OK, 0 rows affected (0.02 sec)
Records: 0  Duplicates: 0  Warnings: 0

mysql> select * from t3;
+------+------+
| id   | name |
+------+------+
|    1 | NULL |
+------+------+
1 row in set (0.01 sec)

mysql> insert into t3 values (2,'B');
Query OK, 1 row affected (0.01 sec)

mysql> select * from t3;
+------+------+
| id   | name |
+------+------+
|    1 | NULL |
|    2 | B    |
+------+------+
2 rows in set (0.00 sec)

mysql> insert into t3 values (3,'D');
ERROR 1265 (01000): Data truncated for column 'name' at row 1
mysql> select * from t3;
+------+------+
| id   | name |
+------+------+
|    1 | NULL |
|    2 | B    |
+------+------+
2 rows in set (0.00 sec)
```

### Steps to Reproduce

```Markdown
create database if not exists test;
use test;
drop table if exists t3;
create table t3 (id int);
insert into t3 values (1);
alter table t3 add column name enum ('A','B','C'); -- Expected: ok
show create table t3;
desc t3;
select * from t3;
insert into t3 values (2,'B'); -- Expected: ok
insert into t3 values (3,'D'); -- Expected: ERROR 1265 (01000): Data truncated for column 'name' at row 1
select * from t3;
```

### Additional information

Same priority as #15076 since from the same source.

**Comment by YANGGMM** [comment]
``` mysql
create table t3 (id int);
insert into t3 values (1);
alter table t3 add column name enum ('A','B','C');
```

**Comment by aronchanisme** [comment]
Fixed in main-`74464bab5`. @Ariznawlll Hi, could you please kindly help add some bvt cases? See 'steps to reproduce' for more info.

```bash
aronchen@momac mo % mo_ctl get_cid 
2024-04-02 11:54:37.376 UTC+0800    [INFO]    Try get mo commit id
commit 74464bab51abe652c7cf77eb70ab45d66f437abb
Author: YANGGMM <www.yangzhao123@gmail.com>
Date:   Tue Apr 2 11:32:36 2024 +0800

    support create table like stmt (#15232)
    
    support create table like stmt
    
    Approved by: @iamlinjunhong, @ouyuanning, @heni02
2024-04-02 11:54:37.444 UTC+0800    [INFO]    Get commit id succeeded
aronchen@momac mo % mo_ctl get_branch
2024-04-02 11:54:40.220 UTC+0800    [INFO]    Try get mo branch
2024-04-02 11:54:40.268 UTC+0800    [INFO]    Get branch succeeded, current branch: main
aronchen@momac mo % 
aronchen@momac mo % mo_ctl connect
2024-04-02 11:56:22.901 UTC+0800    [INFO]    Checking connectivity
2024-04-02 11:56:22.964 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 331
Server version: 8.0.30-MatrixOne-v1.1.2 MatrixOne

Copyright (c) 2000, 2023, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 74464bab5     |
+---------------+
1 row in set (0.00 sec)

mysql> create table t3 (id int);
ERROR 1046 (HY000): not connect to a database
mysql> create database if not exists test;
Query OK, 1 row affected (0.01 sec)

mysql> use test;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> drop table if exists t3;
Query OK, 0 rows affected (0.00 sec)

mysql> create table t3 (id int);
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t3 values (1);
Query OK, 1 row affected (0.01 sec)

mysql> alter table t3 add column name enum ('A','B','C'); -- Expected: ok
Query OK, 0 rows affected (0.02 sec)

mysql> show create table t3;
+-------+------------------------------------------------------------------------------------+
| Table | Create Table                                                                       |
+-------+------------------------------------------------------------------------------------+
| t3    | CREATE TABLE `t3` (
`id` INT DEFAULT NULL,
`name` ENUM('A','B','C') DEFAULT NULL
) |
+-------+------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> desc t3;
+-------+-------------------+------+------+---------+-------+---------+
| Field | Type              | Null | Key  | Default | Extra | Comment |
+-------+-------------------+------+------+---------+-------+---------+
| id    | INT(32)           | YES  |      | NULL    |       |         |
| name  | ENUM('A','B','C') | YES  |      | NULL    |       |         |
+-------+-------------------+------+------+---------+-------+---------+
2 rows in set (0.01 sec)

mysql> select * from t3;
+------+------+
| id   | name |
+------+------+
|    1 | NULL |
+------+------+
1 row in set (0.00 sec)

mysql> insert into t3 values (2,'B'); -- Expected: ok
Query OK, 1 row affected (0.00 sec)

mysql> insert into t3 values (3,'D'); -- Expected: ERROR 1265 (01000): Data truncated for column 'name' at row 1
ERROR 20101 (HY000): internal error: convert to MySQL enum failed: item D is not in enum [A B C]
mysql> select * from t3;
+------+------+
| id   | name |
+------+------+
|    1 | NULL |
|    2 | B    |
+------+------+
2 rows in set (0.00 sec)
```

**Comment by aronchanisme** [comment]
test result on 1.1-dev(`8cd93a370`):
```bash
aronchen@momac mo % mo_ctl get_branch
2024-04-02 12:24:19.875 UTC+0800    [INFO]    Try get mo branch
2024-04-02 12:24:19.937 UTC+0800    [INFO]    Get branch succeeded, current branch: 1.1-dev
aronchen@momac mo % mo_ctl get_cid
2024-04-02 12:24:21.625 UTC+0800    [INFO]    Try get mo commit id
commit 8cd93a370a61f666fa5ffcb5efadd304f24bd7aa
Author: brown <endeavorjia@gmail.com>
Date:   Tue Apr 2 11:35:18 2024 +0800

    CI: fix 1.1 dev ci (#15281)
2024-04-02 12:24:21.687 UTC+0800    [INFO]    Get commit id succeeded
aronchen@momac mo % mo_ctl connect
2024-04-02 12:24:23.973 UTC+0800    [INFO]    Checking connectivity
2024-04-02 12:24:24.047 UTC+0800    [ERROR]    Connect failed
aronchen@momac mo % mo_ctl connect
2024-04-02 12:26:02.485 UTC+0800    [INFO]    Checking connectivity
2024-04-02 12:26:02.547 UTC+0800    [INFO]    Ok, connecting for user ... 
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 23
Server version: 8.0.30-MatrixOne-v1.1.2 MatrixOne

Copyright (c) 2000, 2023, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| 8cd93a370     |
+---------------+
1 row in set (0.00 sec)

mysql> create database if not exists test;
Query OK, 1 row affected (0.01 sec)

mysql> use test;
Database changed
mysql> drop table if exists t3;
Query OK, 0 rows affected (0.00 sec)

mysql> create table t3 (id int);
Query OK, 0 rows affected (0.01 sec)

mysql> insert into t3 values (1);
Query OK, 1 row affected (0.01 sec)

mysql> alter table t3 add column name enum ('A','B','C'); -- Expected: ok
Query OK, 0 rows affected (0.02 sec)

mysql> show create table t3;
+-------+------------------------------------------------------------------------------------+
| Table | Create Table                                                                       |
+-------+------------------------------------------------------------------------------------+
| t3    | CREATE TABLE `t3` (
`id` INT DEFAULT NULL,
`name` ENUM('A','B','C') DEFAULT NULL
) |
+-------+------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> desc t3;
+-------+-------------------+------+------+---------+-------+---------+
| Field | Type              | Null | Key  | Default | Extra | Comment |
+-------+-------------------+------+------+---------+-------+---------+
| id    | INT(32)           | YES  |      | NULL    |       |         |
| name  | ENUM('A','B','C') | YES  |      | NULL    |       |         |
+-------+-------------------+------+------+---------+-------+---------+
2 rows in set (0.01 sec)

mysql> select * from t3;
+------+------+
| id   | name |
+------+------+
|    1 | NULL |
+------+------+
1 row in set (0.01 sec)

mysql> insert into t3 values (2,'B'); -- Expected: ok
Query OK, 1 row affected (0.01 sec)

mysql> insert into t3 values (3,'D'); -- Expected: ERROR 1265 (01000): Data truncated for column 'name' at row 1
ERROR 20101 (HY000): internal error: convert to MySQL enum failed: item D is not in enum [A B C]
mysql> select * from t3;
+------+------+
| id   | name |
+------+------+
|    1 | NULL |
|    2 | B    |
+------+------+
2 rows in set (0.00 sec)
```
